pkg/lib-volume-populator/populator-machinery/controller.go

```diff
@@ -819,104 +821,66 @@ func (c *controller) updatePvc(ctx context.Context, pvc *corev1.PersistentVolume
 	populatorKind := pvc.Spec.DataSourceRef.Kind
-	var diskRegex = regexp.MustCompile(fmt.Sprintf(`volume_populators_%s\{%s=\"([0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12})"\} (\d{1,3}.*)`,
-		populatorToResource[populatorKind].regexKey,
-		populatorToResource[populatorKind].storageResourceKey))
+	importRegExp := regexp.MustCompile("progress\\{ownerUID=\"" + string(pvc.UID) + "\"\\} (\\d+\\.?\\d*)")
 	url := fmt.Sprintf("http://%s:2112/metrics", podIP)
 	resp, err := http.Get(url)
 	if err != nil {
 		if strings.Contains(err.Error(), "connection refused") {
 			return nil
 		}
-		c.recorder.Eventf(pvc, corev1.EventTypeWarning, reasonPopulatorProgress, "Failed to get url: %s, err: %w", url, err)
+		klog.Warning(err)
 		return nil
 	}
 
 	defer resp.Body.Close()
 	body, err := io.ReadAll(resp.Body)
 
 	if err != nil {
-		c.recorder.Eventf(pvc, corev1.EventTypeWarning, reasonPopulatorProgress, "Failed to read response, error: %w", url, err)
+		klog.Warning(err)
 	}
 
-	matches := diskRegex.FindAllStringSubmatch(string(body), -1)
-	if matches == nil {
-		c.recorder.Eventf(pvc, corev1.EventTypeWarning, reasonPopulatorProgress, "Failed to find matches, regex: %s", diskRegex)
+	match := importRegExp.FindStringSubmatch(string(body))
+	if match == nil {
+		klog.Warning("Failed to find matches, regex: ", importRegExp)
 		return nil
 	}
 
-	for _, match := range matches {
-		imageID := match[1]
-		progress, err := strconv.ParseFloat(string(match[2]), 64)
-		if err != nil {
-			c.recorder.Eventf(pvc, corev1.EventTypeWarning, reasonPopulatorProgress, "Could not convert progress: %w", err)
-			break
-		}
-
-		gvr := schema.GroupVersionResource{
-			Group:    *pvc.Spec.DataSourceRef.APIGroup,
-			Version:  "v1beta1",
-			Resource: populatorToResource[populatorKind].resource,
-		}
-
-		latestPopulator, err := c.dynamicClient.Resource(gvr).Namespace(pvc.Namespace).Get(context.TODO(), cr.GetName(), metav1.GetOptions{})
-		if err != nil {
-			c.recorder.Eventf(pvc, corev1.EventTypeWarning, reasonPopulatorProgress, "Failed to get CR for kind: %s error: %w", populatorKind, err)
-			break
-		}
-
-		var updatedPopulator *unstructured.Unstructured
-		switch populatorKind {
-		case v1beta1.OpenstackVolumePopulatorKind:
-			updatedPopulator, err = createUpdatedOpenstackPopulatorProgress(int64(progress), latestPopulator)
-		case v1beta1.OvirtVolumePopulatorKind:
-			updatedPopulator, err = updateOvirtPopulatorProgress(int64(progress), latestPopulator)
-		default:
-			c.recorder.Eventf(pvc, corev1.EventTypeWarning, reasonPopulatorProgress, "Unsupported populator kind: %s", populatorKind)
-		}
-		if err != nil {
-			c.recorder.Eventf(pvc, corev1.EventTypeWarning, reasonPopulatorProgress, "Failed to updated CR for kind: %s error: %w", populatorKind, err)
-			break
-		}
+	progress, err := strconv.ParseFloat(string(match[1]), 64)
+	if err != nil {
+		klog.Warning("Could not convert progress: ", err)
```

Comment:
do you intentionally not returning here and when err != nil below?

Reply:
added